src/Microsoft.DotNet.Build.Tasks/GetNetCoreAppVersionsFromFile.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GetNetCoreAppVersionsFromFile : Task
+    {
+        [Required]
+        public string PathToVersionsFile { get; set; }
+
+        [Output]
+        public string CorefxHash { get; set; }
```

Comment:
I see, it shouldn't be more difficult to consume, so I'll change it to be generic. I don't have a commit yet with how to consume it, as I'm working on a change that will calculate the path to this versions file as since it is an SDK project that restores it, the package won't be restored to the `PackagesDir`. Once this is merged, I'll send out the corefx PR.